presto-main/src/main/java/com/facebook/presto/server/protocol/QueuedStatementResource.java

```diff
@@ -380,7 +392,7 @@ public synchronized QueryResults getInitialQueryResults(UriInfo uriInfo, String
                             uriInfo,
                             xForwardedProto,
                             DispatchInfo.queued(NO_DURATION, NO_DURATION));
-                    return immediateFuture(Response.ok(queryResults).build());
+                    return immediateFuture(withCompressionEnabled(Response.ok(queryResults), compressionEnabled).build());
```

Comment:
I understand that we are trying to reduce duplicate code, but passing `compressionEnabled ` to `withCompressionEnabled` is a bit weird and sounds like this should be done outside.

Reply:
Maybe: `withCompressionConfiguration(Response.Builder builder, boolean compressionDisabled)` ?

Reply:
Yeah that seems better :D